config.md

```diff
@@ -28,10 +28,12 @@ For example, if a configuration is compliant with version 1.1 of this specificat
 
 **`root`** (object, REQUIRED) specifies the container's root filesystem.
 
-* **`path`** (string, REQUIRED) Specifies the path to the root filesystem for the container.
+* **`path`** (string, OPTIONAL) Specifies the path to the root filesystem for the container.
   The path is either an absolute path or a relative path to the bundle.
-  On Linux, for example, with a bundle at `/to/bundle` and a root filesystem at `/to/bundle/rootfs`, the `path` value can be either `/to/bundle/rootfs` or `rootfs`.
-  A directory MUST exist at the path declared by the field.
+  On Linux, for example, with a bundle at `/to/bundle` and a root filesystem at `/to/bundle/rootfs`, the `path` value can be either `/to/bundle/rootfs` or `rootfs`. 
+  On Windows, for Windows Server Containers, this field is REQUIRED. For Hyper-V Containers, this field MUST NOT be supplied.
```

Comment:
For writing validation tooling, how should runtime-tools distinguish between a Windows Server config and a Hyper-V config?

This change will also need some knock ons:

* Adjusting [the][1] [bundle][2] [requirements][3] to only require the root filesystem directory when `root.path` is set.  I still don't see a need for that bundle requirement (last discussed in #469) and think that the `config.md` requirement (which you preserve in this PR) for “A directory MUST exist at the path declared by the field” is sufficient for sanity.  But if you can't get consensus around something like that, you'll still need to do something to `bundle.md`, since the current wording there makes it impossible to leave `root.path` unset.
* An `omitempty` [in the Go type][4].
* Dropping the `path` requirement [from the JSON Schema][5].

[1]: https://github.com/opencontainers/runtime-spec/blame/559acdf05cd4dd0b57d720a3981147597d249280/bundle.md#L11
[2]: https://github.com/opencontainers/runtime-spec/blame/559acdf05cd4dd0b57d720a3981147597d249280/bundle.md#L17-L19
[3]: https://github.com/opencontainers/runtime-spec/blame/559acdf05cd4dd0b57d720a3981147597d249280/bundle.md#L21-L21
[4]: https://github.com/opencontainers/runtime-spec/blob/559acdf05cd4dd0b57d720a3981147597d249280/specs-go/config.go#L99
[5]: https://github.com/opencontainers/runtime-spec/blob/559acdf05cd4dd0b57d720a3981147597d249280/schema/config-schema.json#L62-L64